teamfoundationclient.cpp

```diff
@@ -382,3 +405,26 @@ void TeamFoundationClient::addRecursive(QStringList &arguments, const QString &p
     if (fileInfo.isDir())
         arguments << QLatin1String("/recursive");
 }
+
+void TeamFoundationClient::configurationChanged()
+{
+    if (m_plugin->settings().binaryPath().isEmpty()) {
+        m_tfVersion = TfVersion_None;
+    } else {
+        TeamFoundationResponse response = runTf(QStringLiteral("C:/"), QStringList(), SuppressCompletely);
+        QRegularExpression re(QStringLiteral("Version (\\d+)"));
+        QRegularExpressionMatch match = re.match(response.standardOut);
+        if (match.hasMatch()) {

```

Comment:
If there is no match, nothing is done to m_tfVersion - why not always default initialize it to TfVersion_None at the begining of the function and overwrite it if there is a match?